src/cards/stats-card.js

```diff
@@ -203,13 +184,20 @@ const renderStatsCard = (stats = {}, options = { hide: [] }) => {
     return measureText(custom_title ? custom_title : i18n.t("statcard.title"));
   };
 
-  const width = hide_rank
-    ? clampValue(
-        50 /* padding */ + calculateTextWidth() * 2,
-        270 /* min */,
-        Infinity,
-      )
-    : 495;
+  /*
+    When hide_rank=true, the minimum card width is 270 px + the title length and padding.
+    When hide_rank=false, the minimum card_width is 340 px + the icon width (if show_icons=true).
+    Numbers are picked by looking at existing dimensions on production.
+  */
+  const iconWidth = show_icons ? 16 : 0;
+  const minCardWidth = hide_rank
+    ? clampValue(50 /* padding */ + calculateTextWidth() * 2, 270, Infinity)
+    : 340 + iconWidth;
+  const defaultCardWidth = hide_rank ? 270 : 495;
```

Comment:
@postatum, @anuraghazra Maybe we should also add the `iconWidth` to the `minwidth` of the card when `hide_rank=true`. By doing this the behaviour for the `hide_icons` option is similar to the case where `hide_rank=false`.

```js
const defaultCardWidth = hide_rank ? 270 + show_icons : 495;
```